skopt/space.py

```diff
@@ -351,22 +351,12 @@ def transform(self, X):
             The transformed samples.
         """
         # Pack by dimension
-        columns = []
-        for dim in self.dimensions:
-            columns.append([])
+        X = np.asarray(X)
```

Comment:
Is this always safe? My concern is that this transforms `X` into an array, which comes with casting. Shall we maybe set the dtype to `np.object` to avoid some downcasting effects?